extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -472,7 +473,15 @@ private ByteBuffer getFullFragment() throws ExecutionException, InterruptedExcep
             // Giving our own get() a bit of overhead - the clients should already be constructed with appropriate
             // apiCallTimeout.
             final long readNanos = instructions.readTimeout().plusMillis(100).toNanos();
-            final ByteBuffer result = consumerFuture.get(readNanos, TimeUnit.NANOSECONDS);
+            final Boolean isComplete = consumerFuture.get(readNanos, TimeUnit.NANOSECONDS);
+            if (!isComplete) {
+                throw new UncheckedDeephavenException(String.format("Failed to complete request %s", requestStr()));
+            }
+            final ByteBuffer result = get();
+            if (result == null) {
+                throw new IllegalStateException(
+                        String.format("Failed to acquire buffer after completion, %s", requestStr()));
```

Comment:
This seems wrong. We need to re-try, sure, but unless we know someone still has us in local cache, this is a valid outcome.

Reply:
I am a bit confused about this one, let us discuss this because right now, I think this case is not handled properly.